x/subkeys/internal/keeper/keeper.go

```diff
@@ -0,0 +1,136 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/subkeys/exported"
+	"github.com/cosmos/cosmos-sdk/x/subkeys/internal/types"
+)
+
+type Keeper struct {
+	cdc      *codec.Codec
+	storeKey sdk.StoreKey
+}
+
+// NewKeeper creates a DelegationKeeper
+func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey) Keeper {
+	return Keeper{cdc: cdc, storeKey: storeKey}
+}
+
+// DelegateFeeAllowance creates a new grant
+func (k Keeper) DelegateFeeAllowance(ctx sdk.Context, grant types.FeeAllowanceGrant) error {
+	store := ctx.KVStore(k.storeKey)
+	key := types.FeeAllowanceKey(grant.Granter, grant.Grantee)
+
+	bz, err := k.cdc.MarshalBinaryBare(grant)
+	if err != nil {
+		return err
+	}
+	store.Set(key, bz)
+	return nil
+}
+
+// RevokeFeeAllowance removes an existing grant
+func (k Keeper) RevokeFeeAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.FeeAllowanceKey(granter, grantee)
+	store.Delete(key)
+}
+
+// GetFeeAllowance returns the allowance between the granter and grantee.
+// If there is none, it returns nil, nil.
+// Returns an error on parsing issues
+func (k Keeper) GetFeeAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress) (exported.FeeAllowance, error) {
+	grant, err := k.GetFeeGrant(ctx, granter, grantee)
+	if grant == nil {
+		return nil, err
+	}
+	return grant.Allowance, err
+}
+
+// GetFeeGrant returns entire grant between both accounts
+func (k Keeper) GetFeeGrant(ctx sdk.Context, granter sdk.AccAddress, grantee sdk.AccAddress) (*types.FeeAllowanceGrant, error) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.FeeAllowanceKey(granter, grantee)
+	bz := store.Get(key)
+	if len(bz) == 0 {
+		return nil, nil
+	}
+
+	var grant types.FeeAllowanceGrant
+	err := k.cdc.UnmarshalBinaryBare(bz, &grant)
+	if err != nil {
+		return nil, err
+	}
+	return &grant, nil
+}
+
+// GetAllGranteeFeeAllowances returns a list of all the grants from anyone to the given grantee.
+func (k Keeper) GetAllGranteeFeeAllowances(ctx sdk.Context, grantee sdk.AccAddress) ([]types.FeeAllowanceGrant, error) {
```

Comment:
@ethanfrey where are the iterator methods?

Reply:
Now done